homeassistant/components/synology_dsm/config_flow.py

```diff
@@ -208,7 +208,7 @@ async def async_step_2sa(self, user_input, errors=None):
     def _host_already_configured(self, hostname):
         """See if we already have a host matching user input configured."""
         existing_hosts = {
-            entry.data[CONF_HOST] for entry in self._async_current_entries()
+            entry.data.get(CONF_HOST) for entry in self._async_current_entries()
```

Comment:
This is the case where it defaults to localhost when its missing?

Reply:
The host is actually required, but it was to fix #34954 until we remove this function to check `_macs_already_configured` as agrred here https://github.com/home-assistant/core/issues/34888#issuecomment-622615865

Needed stuffs to get the MAC addresses are on 0.8.0 and updated in this PR #35068, and throught this was a breaking change.

But if you correct the MAC addresses, we actually don't need this PR.

(I hope I made myself clear :smile:)

Reply:
I converted this to a draft so it doesn't merge as we should probably fix the duplicate discovery issue instead

Reply:
Ok, 0.8.0 adds `dsm_api.network.macs` to do so 😉

Closing this PR